chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/UserManagementController.java

```diff
@@ -408,22 +409,36 @@ private String delete(final Long userId)
             throw new UserRetrievalException("Could not find user with id " + userId);
         }
 
-        // delete the atl permissions for that user
-        userPermissionsManager.deleteAllAtlPermissionsForUser(userId);
-        
-        // delete the acb permissions for that user
-        userPermissionsManager.deleteAllAcbPermissionsForUser(userId);
-        
-        // delete the user
-        userManager.delete(toDelete);
+        //If the current user is ROLE_ADMIN or ROLE_ONC, delete the user and access to all ACB and ATLS
+        //If the current user is ROLE_ACB, remove all of the ACBs that the current user has from the target user
+        //If the current user is ROLE_ATL, remove all of the ATLs that the current user has from the target user
+        if (resourcePermissions.isUserRoleAdmin() || resourcePermissions.isUserRoleOnc()) {
+            userManager.delete(toDelete);
+            userPermissionsManager.deleteAllAcbPermissionsForUser(userId);
+            userPermissionsManager.deleteAllAtlPermissionsForUser(userId);
+        } else if (resourcePermissions.isUserRoleAcbAdmin()) {
+            List<CertificationBodyDTO> targetUserAcbs = resourcePermissions.getAllAcbsForUser(userId);
+            for (CertificationBodyDTO dto : resourcePermissions.getAllAcbsForCurrentUser()) {
+                if (isAcbInList(dto, targetUserAcbs)) {
+                    userPermissionsManager.deleteAcbPermission(dto, userId);
+                }
+            }
+        } else if (resourcePermissions.isUserRoleAtlAdmin()) {
```

Comment:
If a user has ROLE_ACB and ROLE_ATL I think this won't work as desired? I know that's super unlikely. But if they do, I think the if/else if's need to be changed around a bit.